### ISSUE TITLE
geth 1.2.2, possible memory leak

### ISSUE THREAD

**Issue by datajerk** [ethereum/go-ethereum#1930]
I can create by doing this:
```
geth & # watch this process memory
while :
do
geth --exec admin.peers attach
done
```
After a while the main geth process running in the background will run out of memory.

**Comment by karalabe** [comment]
Hmm... tried this on latest develop just now, been running for 5 mins now, but the original geth won't budge from +- 150MB memory usage.

**Comment by datajerk** [comment]
Takes hours. Try 24.

**Comment by karalabe** [comment]
Do you have a crash dump from such an out of memory panic? Could help us see what was going on inside the node.

**Comment by datajerk** [comment]
Give me 24 hours.

**Comment by fjl** [comment]
@datajerk did it occur again?

**Comment by datajerk** [comment]
@fjl yes, tried this instead:

```
while :; do geth --exec web3.net.peerCount attach; done
```

But the OS killed it when it exhausted memory--no dump to speak of.

I'll test the latest dev branch.

**Comment by obscuren** [comment]
@datajerk it's only been, potentially, fixed on `develop`

**Comment by obscuren** [comment]
@fjl found the issue. He's working on a patch tomorrow.

**Comment by datajerk** [comment]
Good news.  BTW, I wanted to rule out the js console and was able to reproduce issue faster with JSON RPC:

```
while :; do echo '{"jsonrpc":"2.0","method":"net_peerCount","params":[],"id":1}' | nc -U /root/.ethereum/geth.ipc; echo; done
```

**Comment by obscuren** [comment]
Correct. During initialisation of the IPC connection a service is start up and doesn't shutdown when you close the connection.

**Comment by fjl** [comment]
The fix has been released in geth 1.3.1.
